src/BlockTypes/ProductCategories.php

```diff
@@ -134,15 +136,37 @@ protected function get_container_classes( $attributes = array() ) {
 	 * @return array
 	 */
 	protected function get_categories( $attributes ) {
-		$hierarchical = wc_string_to_bool( $attributes['isHierarchical'] );
-		$categories   = get_terms(
-			'product_cat',
-			[
-				'hide_empty'   => ! $attributes['hasEmpty'],
-				'pad_counts'   => true,
-				'hierarchical' => true,
-			]
-		);
+		$hierarchical  = wc_string_to_bool( $attributes['isHierarchical'] );
+		$children_only = wc_string_to_bool( $attributes['showChildrenOnly'] ) && is_product_category();
+
+		if ( $children_only ) {
+			$term_id    = get_queried_object_id();
+			$categories = get_terms(
+				'product_cat',
+				[
+					'hide_empty'   => ! $attributes['hasEmpty'],
+					'pad_counts'   => true,
+					'hierarchical' => true,
+					'child_of'     => $term_id,
+				]
+			);
+		} else {
+			$categories = get_terms(
+				'product_cat',
+				[
+					'hide_empty'   => ! $attributes['hasEmpty'],
+					'pad_counts'   => true,
+					'hierarchical' => true,
+				]
+			);
+		}
+
+		// If they have $children_only enabled, and the current category is the last child in the hierarchy
+		// we need to show the current category instead of rendering nothing at all.
+		if ( $children_only && ( ! is_array( $categories ) || empty( $categories ) ) ) {
+			$current_category = get_term( $term_id, 'product_cat' );
+			return [ $current_category ];
+		}
```

Comment:
Have alternatives been considered? I'm not saying I don't like this approach, but I think I would have expected it to either display nothing or display the current category + all siblings.

Reply:
Matching the behavior of the existing widget is a solid place to start. That said, I'm not entirely sure I understand the desired outcome, nor the intended use-case.

Any chance you can include some visual for how the block will behave/render in each scenario? For instance, does the current category always show, or will it only show if no other categories exist? It seems like it might be strange to only show the current category with some relatively obscure logic.

Reply:
>Matching the behavior of the existing widget is a solid place to start.

Yeah, I agree. I've just reactivated the classic widget locally and done a side-by-side test which I've screen recorded below. In the below video I'm on the "Accessories" product category which has a "Grandchild" sub-category I created which is last in the hierachy, so this is the only one that shows initially. Going to this category results in it disappearing because there are no more sub-categories. Based on this I'll alter the block features logic to mirror this and if there are any further improvements we want to make we can do so in a follow up PR.

https://github.com/woocommerce/woocommerce-blocks/assets/8639742/d7f09b3b-b817-455f-bf2d-f66300e954f5

Reply:
@Aljullu I've remove the logic in question here https://github.com/woocommerce/woocommerce-blocks/pull/9550/commits/00feab1d63b5e0fe2692abab937b12d3c5746078

Something worth mentioning is that if you add a `Title` block before the `Product Categories List` block then when it renders nothing, the title block will still appear.

We fixed this in other blocks such as the filter blocks but this is because those blocks originally _came with_ a title when inserted, whereas the `Product Categories List` block does not. So there is no relationship between the two blocks.